Makefile

```diff
@@ -60,6 +61,16 @@ ifneq ($(HAS_SERVER),)
 	cd server && env GOOS=windows GOARCH=amd64 $(GO) build $(GO_BUILD_FLAGS) -o dist/plugin-windows-amd64.exe;
 endif
 
+## Builds the server, if it exists, in debug mode for all supported architectures.
+.PHONY: server-debug
+server-debug:
+ifneq ($(HAS_SERVER),)
+	mkdir -p server/dist
+	cd server && env GOOS=darwin GOARCH=amd64 $(GO) build $(GO_BUILD_FLAGS) -gcflags "all=-N -l" -o dist/plugin-darwin-amd64;
+	cd server && env GOOS=linux GOARCH=amd64 $(GO) build $(GO_BUILD_FLAGS) -gcflags "all=-N -l" -o dist/plugin-linux-amd64;
+	cd server && env GOOS=windows GOARCH=amd64 $(GO) build $(GO_BUILD_FLAGS) -gcflags "all=-N -l" -o dist/plugin-windows-amd64.exe;
+endif
```

Comment:
Just wondering if there's a reason for removing the single-arch build logic?

Reply:
I found the behaviour "surprising", in that it assumed I was building and deploying to a debug version to the same machine / architecture. I realize that's intently common for Mattermost stuff, but wondering if there is a potential use case for not doing that?

I had half a mind to formalize an architecture override -- for both production and debug builds -- with an environment variable or some such. Thoughts?

Reply:
Maybe it's more useful in the way server-debug works in IR: build just for your arch, then upload just the server side of the plugin, then reset the plugin. It takes a second or two, so it's useful when developing. I guess we're removing that functionality here?

Reply:
Ah, that makes sense. Let me revisit all of this re: watching.

Reply:
I ended up leaving the build-all-architectures here for the time being, despite restoring `make watch` as described below. Let me know if you think this is a blocker!